src/d3a/d3a_core/simulation.py

```diff
@@ -262,6 +262,8 @@ def _update_and_send_results(self, is_final=False):
             filename = self.export.export_to_zip_file()
             self.redis_connection.write_zip_results(filename)
             self.export.delete_exported_files()
+            if hasattr(self.redis_connection, 'heartbeat'):
+                self.redis_connection.heartbeat.cancel()
```

Comment:
So if the d3a is sending the results, the `RepeatingTimer` gets stopped?
`_update_and_send_results` is called on every market slot. Is this really what you wanted? I thought is should send a heartbeat tick every PULSE_RATE  seconds no matter what.

Reply:
Have a look at L257 at this point of commit, it is when simulation simulation has finished and now `RepeatingTimer ` should stop too

Reply:
This is actually the place where the simulation has actually finished and is preparing the zip files. Therefore I think it is a correct place to have it there, although this will probably be refactored shortly after we skip generating the archive with the csv files in the d3a.